src/flow/src/expr/scalar.rs

```diff
@@ -47,6 +57,8 @@ impl TypedExpr {
     /// expand multi-value expression to multiple expressions with new indices
     ///
     /// Currently it just mean expand `TumbleWindow` to `TumbleWindowFloor` and `TumbleWindowCeiling`
+    ///
+    /// TODO(discord9): test if nested reduce combine with df scalar function would cause problem
```

Comment:
It's better to test it right now. Do we find any way to test the flow in a black box?

Reply:
Pending this pull request in sqlness to add those tests(because flow test need to `sleep` to wait for results to be updated): https://github.com/CeresDB/sqlness/pull/67